VHbbAnalysis/Heppy/python/AddRegression.py

```diff
@@ -0,0 +1,92 @@
+from PhysicsTools.Heppy.analyzers.core.Analyzer import Analyzer
+from VHbbAnalysis.Heppy.vhbbobj import ptRel
+from math import *
+import ROOT
+import array
+
+class AddRegression( Analyzer ):
+
+
+
+   def declareHandles(self):
+        reader = ROOT.TMVA.Reader()
+        self.Jet_pt =array.array('f',[0])
+        self.Jet_rawPt = array.array('f',[0])
+        self.rho = array.array('f',[0])
+        self.Jet_eta = array.array('f',[0])
+        self.Jet_mt = array.array('f',[0])
+        self.Jet_leadTrackPt = array.array('f',[0])
+        self.Jet_leptonPtRel = array.array('f',[0])
+        self.Jet_leptonPt =  array.array('f',[0])
+        self.Jet_leptonDeltaR = array.array('f',[0])
+        self.Jet_chEmEF = array.array('f',[0])
+        self.Jet_chHEF = array.array('f',[0])
+        self.Jet_neHEF = array.array('f',[0])
+        self.Jet_neEmEF = array.array('f',[0])
+        self.Jet_chMult = array.array('f',[0])
+        self.Jet_vtxPt = array.array('f',[0])
+        self.Jet_vtxMass = array.array('f',[0])
+        self.Jet_vtx3dL = array.array('f',[0])
+        self.Jet_vtxNtrk = array.array('f',[0])
+        self.Jet_vtx3deL = array.array('f',[0])
+        reader.AddVariable("Jet_pt",self.Jet_pt)
+        reader.AddVariable("Jet_rawPt",self.Jet_rawPt)
+        reader.AddVariable("rho",self.rho)
+        reader.AddVariable("Jet_eta",self.Jet_eta)
+        reader.AddVariable("Jet_mt",self.Jet_mt)
+        reader.AddVariable("Jet_leadTrackPt",self.Jet_leadTrackPt)
+        reader.AddVariable("Jet_leptonPtRel",self.Jet_leptonPtRel)
+        reader.AddVariable("Jet_leptonPt",self.Jet_leptonPt)
+        reader.AddVariable("Jet_leptonDeltaR",self.Jet_leptonDeltaR)
+        reader.AddVariable("Jet_chEmEF",self.Jet_chEmEF)
+        reader.AddVariable("Jet_chHEF",self.Jet_chHEF)
+        reader.AddVariable("Jet_neHEF",self.Jet_neHEF)
+        reader.AddVariable("Jet_neEmEF",self.Jet_neEmEF)
+        reader.AddVariable("Jet_chMult",self.Jet_chMult)
+        reader.AddVariable("Jet_vtxPt",self.Jet_vtxPt)
+        reader.AddVariable("Jet_vtxMass",self.Jet_vtxMass)
+        reader.AddVariable("Jet_vtx3dL",self.Jet_vtx3dL)
+        reader.AddVariable("Jet_vtxNtrk",self.Jet_vtxNtrk)
+        reader.AddVariable("Jet_vtx3deL",self.Jet_vtx3deL)
+        if event.Vtype<2 :
+                reader.BookMVA("jet0Regression","../weights/Zll_weights_phys14.xml")
+        elif event.Vtype <4 :
+                reader.BookMVA("jet0Regression","../weights/Wln_weights_phys14.xml")
+        elif event.Vtype ==4:
+                reader.BookMVA("jet0Regression","../weights/Znn_weights_phys14.xml")
+        self.reader=reader
+   def process(self, event):
+	self.readCollections( event.input )
+        self.rho[0] = event.rho
+        for j in event.cleanJets :
```

Comment:
mhh it doesn't make sense and I guess takes too much time to run on all cleanJets.
What I meant with factorizing wasn't to make a module but rather to have a python file to import in Vhbb and just call a regression function from there, only on the proper jets.
In any case if you run in a separate analyzer, why not regressing only the higgs jets?